js/chat/chat.ts

```diff
@@ -60,20 +61,19 @@ class ChatMessage extends LightElement {
   @property() content = "...";
   @property() content_type: ContentType = "markdown";
   @property({ type: Boolean, reflect: true }) streaming = false;
+  @property() icon = "";
 
   render() {
-    const noContent = this.content.trim().length === 0;
-    const defaultIcon = noContent ? ICONS.dots_fade : ICONS.robot;
+    let msg_icon = ICONS.dots_fade;
 
-    // Check if there's an existing message-icon element
-    const userIcon = this.querySelector(".message-icon");
-    const icon =
-      userIcon && !noContent
-        ? userIcon
-        : html`<div class="message-icon">${unsafeHTML(defaultIcon)}</div>`;
+    // Show dots until we have content
+    const hasContent = this.content.trim().length !== 0;
+    if (hasContent) {
+      msg_icon = this.icon || ICONS.robot;
+    }
```

Comment:
I'd rather use the ternary here so that `icon` is set once with `const`.

```js
const hasContent = this.content.trim().length !== 0;
const icon = hasContent ? this.icon || ICONS.robot : ICONS.dots_fade;
```